internal/safebuffer/safebuffer.go

```diff
@@ -0,0 +1,88 @@
+package safebuffer
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"sync"
+
+	"github.com/joshuarubin/teleport-job-worker/internal/safebuffer/safereader"
+)
+
+// Buffer is a goroutine safe buffer that ingests data as an io.Writer and
+// provides io.ReadClosers to reply and continue to stream incoming data.
+type Buffer struct {
+	mu      sync.RWMutex
+	buf     bytes.Buffer
+	jobDone <-chan struct{}
+	chans   map[chan []byte]<-chan struct{}
```

Comment:
Should this be a slice of Channels instead?

Reply:
The buffer needs to know if a connected reader has been closed so it doesn't attempt to send to a closed channel causing a panic. Are you suggesting using something like this instead?:
```go
[]struct{
    DataCh chan []byte
    DoneCh ←chan struct{}
}
```

Reply:
I meant `[]*safereader.Channels`.